### PR TITLE
Random ship events: Phase 1

### PR DESCRIPTION
**Pull request by Qustinnus**
:cl: Floyd
add: random events that appear once per planet
add: ruin events
add: combat events
remove: removes static spawned ships at planets
/:cl:

### CODE REVIEW

**Comment by Qustinnus** [comment]
need help from mappers look at xeno_hive.dmm for an example

**Comment by Vivalas** [comment]
Screenshots / explanation of events?

**Comment by Qustinnus** [comment]
currently you fly into an unexplored planet (besides capitals) and it triggers an event, this can be a pirate ship or a ruin for the crew to explore. dont really have screenshots of much as the ruins are just placeholder.

**Comment by Qustinnus** [comment]
fuck u cunt :^))))))))

**Comment by Qustinnus** [comment]
load it up and youll see

**Comment by Vivalas** [comment]
Hey look you're not a twink anymore good job--

also screenshots, maintainers can't be assed to actually load anything, remember? That good :ok_hand: life.

**Comment by Qustinnus** [comment]
what will i screenshot literaly everything is behind the scenes reee.

**Comment by Vivalas** [comment]
someone do something ree, I don't trust myself

**Comment by TMTIME** [comment]
I'll take a look this week

File: _maps/ship_encounters/events/ruins/factory_impact.dmm [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+"cR" = (/obj/machinery/door/airlock/external,/turf/open/floor/plating,/area/space)
+"cS" = (/turf/closed/indestructible/rock,/area/space)
+"cT" = (/obj/structure/fans/tiny,/obj/machinery/door/airlock/external,/turf/open/floor/plating,/area/space)
+"cU" = (/turf/closed/wall/r_wall/rust,/area/no_entry)
+"cV" = (/obj/machinery/door/airlock/external,/turf/open/floor/engine,/area/no_entry)
+"cW" = (/turf/open/floor/engine,/area/no_entry)
+"cX" = (/obj/machinery/light/small{dir = 2},/turf/open/floor/engine,/area/no_entry)
+"cY" = (/obj/machinery/door/airlock/external,/obj/structure/fans/tiny,/obj/machinery/door/airlock/external,/obj/structure/fans/tiny,/turf/open/floor/engine,/area/no_entry)
+"cZ" = (/obj/structure/grille,/obj/structure/window/reinforced/fulltile,/obj/structure/fans/tiny,/obj/structure/grille,/obj/structure/window/reinforced/fulltile,/obj/structure/fans/tiny,/turf/open/floor/plating,/area/no_entry)
+"da" = (/obj/effect/landmark/dock_spawn/ruin,/turf/open/space,/area/space)
```
Comment (TMTIME) [inline _maps/ship_encounters/events/ruins/factory_impact.dmm]: Dock spawns are too close to the ruins, crashing the shuttles with no survivors.

Best to place them slightly north of ruins bc the dock landmark is on the south of shuttles

File: code/modules/ftl/events/combat_events.dm [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	description = "An assortment of pirate ships is spotted in orbit of the planet, It seems like they stole these ships somewhere."
+	faction = FTL_PIRATE
+	rarity = UNCOMMON_EVENT
+	var/amount = 3
+	ships_to_spawn = list()
+
+/datum/ftl_event/combat/random/New()
+	amount = rand(1, 3)
+	for(var/i = 1 to amount)
+		ships_to_spawn = pickweight(SSship.ship_weight_list)
```
[the remark this answers is not in the document]
Comment (Qustinnus) [reply code/modules/ftl/events/combat_events.dm]: lmao oops

File: code/controllers/subsystem/starmap.dm [hunk 3]
```diff
@@ -427,12 +461,6 @@ SUBSYSTEM_DEF(starmap)
 
 		faction.systems = sortList(faction.systems,/proc/cmp_danger_dsc) //sorts systems in descending order based on danger level
 
-		var/list/h_list = SSship.faction2list(faction.cname,1)
-		for(var/datum/starship/S in h_list)
-			var/datum/starship/ship_spawned = SSship.create_ship(S,faction.cname,faction.capital)
-			ship_spawned.mission_ai = new /datum/ship_ai/guard
-			ship_spawned.mission_ai:assigned_system = faction.capital
```
Comment (TMTIME) [inline code/controllers/subsystem/starmap.dm]: muh shitty boss ship code

Comment (Qustinnus) [reply code/controllers/subsystem/starmap.dm]: its really gross

File: _maps/ship_encounters/events/ruins/factory_impact.dmm (hunk 1 above)
Comment (Qustinnus) [reply _maps/ship_encounters/events/ruins/factory_impact.dmm]: not sure how to place them, could u take a peek?

Comment (Qustinnus) [reply _maps/ship_encounters/events/ruins/factory_impact.dmm]: @TMTIME

**Comment by crossedfall** [comment]
Closing in favor of 2.0

**Pull request closed without merging**